### ISSUE TITLE
exchange icon instead of attack on enemy hero

### ISSUE THREAD

**Issue by kambala-decapitator** [vcmi/vcmi#790]
purple is my enemy, but once hovering his hero showed exchange icon instead of attack, happened only once and only with one purple hero: https://imgur.com/a/2rt4PJ2

**Comment by kambala-decapitator** [comment]
might be a duplicate of #800
